File: mini_computer_api/sdk_lcm/pyth/listener.py
import lcm
from exlcm import sdk_reqs

def my_handler(channel, data):
    msg = sdk_reqs.decode(data)
    print("Received message on channel \"%s\"" % channel)
    print("   init sdk      = %s" % msg.init_sdk)
    print("   cam conn      = %s" % msg.cam_conn)
    print("   cam status    = %s" % msg.cam_status)
    print("   cam trigger   = %s" % msg.cam_trigger)
    print("   cam disconn   = %s" % msg.cam_disconn)
    print("   release sdk   = %s" % msg.release_sdk)

lcm_subscriber = lcm.LCM()
subscription = lcm_subscriber.subscribe("TO_SDK", my_handler)

try:
    while True:
        lcm_subscriber.handle()
except KeyboardInterrupt:
    pass

File: mini_computer_api/sdk_lcm/cpp/listener.cpp
#include <stdio.h>
#include <lcm/lcm-cpp.hpp>
#include "exlcm/sdk_resp.hpp"

class Handler {
  public:
    ~Handler() {}
    void handleMessage(const lcm::ReceiveBuffer *rbuf, const std::string &chan,
                       const exlcm::sdk_resp *msg)
    {
        printf("Received message on channel \"%s\":\n", chan.c_str());
        printf("  init     = %d\n", msg->resp_init);
        printf("  conn     = %d\n", msg->resp_conn);
        printf("  status   = %d\n", msg->resp_status);
    }
};

int main(int argc, char **argv)
{
    lcm::LCM lcm_subscriber;

    if (!lcm_subscriber.good())
        return 1;

    Handler handlerObject;
    lcm_subscriber.subscribe("FROM_SDK", &Handler::handleMessage, &handlerObject);

    while (0 == lcm_subscriber.handle()) {
        // Do nothing
    }

    return 0;
}